### PR TITLE
remove output warning in MSBuild

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Fix: Remove output warnings in MSBuild helper.
Docs: Omit

### CODE REVIEW

**Comment by jgsogo** [comment]
Ready to merge, just wondering if we should merge it to `develop` or to `release/1.12.3` if we are going to release that minor one.

File: conans/client/build/msbuild.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 from conans.errors import ConanException
 from conans.model.conan_file import ConanFile
 from conans.model.version import Version
+from conans.tools import vcvars_command as tools_vcvars_command
```
Comment (memsharded) [inline conans/client/build/msbuild.py]: Didn't we say that conan does NOT import from ``conans.tools`` and that is exclusively for users/recipes? Besides 2 remaining things, that should be removed, that is the way it is now.

Comment (jgsogo) [reply conans/client/build/msbuild.py]: @danimtb pointed out that **it is much more important not to expose an `output` argument in the public API/tool**. I agree with him, the signature for `MSBuild::get_version()` cannot require the user to provide an `output`.

Comment (memsharded) [reply conans/client/build/msbuild.py]: The problem is that MSBuild.get_version() prints that ugly warning, that users cannot fix.
Also the MSBuild binary_log feature, uses internally this functionality, printing the ugly warning message, and that cannot be fixed either by the user. 

This has to be fixed, outputting warnings that the user cannot fix doesn't make sense, no matter how "clean" our api is.

Comment (jgsogo) [reply conans/client/build/msbuild.py]: The _patch_ by @danimtb avoids that warning (secondary goal) and do not expose the `output` argument (main goal).

**Pull request merged**